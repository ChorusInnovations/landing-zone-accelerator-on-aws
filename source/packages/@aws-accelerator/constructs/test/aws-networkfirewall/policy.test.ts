/**
 *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';

import { FirewallPolicyProperty, NetworkFirewallPolicy } from '../../lib/aws-networkfirewall/policy';

const testNamePrefix = 'Construct(NetworkFirewallPolicy): ';

//Initialize stack for resource configuration test
const stack = new cdk.Stack();

const firewallPolicy: FirewallPolicyProperty = {
  statelessDefaultActions: ['aws:forward_to_sfe'],
  statelessFragmentDefaultActions: ['aws:forward_to_sfe'],
  statefulEngineOptions: 'STRICT_ORDER',
  statefulRuleGroupReferences: [
    {
      priority: 123,
      resourceArn: 'arn:aws:network-firewall:us-east-1:222222222222:stateful-rulegroup/TestGroup',
    },
  ],
};

new NetworkFirewallPolicy(stack, 'TestPolicy', {
  firewallPolicy: firewallPolicy,
  name: 'TestFirewallPolicy',
  tags: [],
});

/**
 * Network Firewall construct test
 */
describe('Network Firewall Policy', () => {
  /**
   * Number of Network Firewall Policy test
   */
  test(`${testNamePrefix} Network firewall count test`, () => {
    cdk.assertions.Template.fromStack(stack).resourceCountIs('AWS::NetworkFirewall::FirewallPolicy', 1);
  });

  /**
   * Network firewall resource configuration test
   */
  test(`${testNamePrefix} Network firewall policy resource configuration test`, () => {
    cdk.assertions.Template.fromStack(stack).templateMatches({
      Resources: {
        TestPolicyCC05E598: {
          Type: 'AWS::NetworkFirewall::FirewallPolicy',
          Properties: {
            FirewallPolicy: {
              StatefulEngineOptions: {
                RuleOrder: 'STRICT_ORDER',
              },
              StatefulRuleGroupReferences: [
                {
                  Priority: 123,
                  ResourceArn: 'arn:aws:network-firewall:us-east-1:222222222222:stateful-rulegroup/TestGroup',
                },
              ],
              StatelessDefaultActions: ['aws:forward_to_sfe'],
              StatelessFragmentDefaultActions: ['aws:forward_to_sfe'],
            },
            FirewallPolicyName: 'TestFirewallPolicy',
            Tags: [{ Key: 'Name', Value: 'TestFirewallPolicy' }],
          },
        },
      },
    });
  });
});
