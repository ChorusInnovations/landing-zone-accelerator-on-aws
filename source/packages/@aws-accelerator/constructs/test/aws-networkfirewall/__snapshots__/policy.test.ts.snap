// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network Firewall Policy Construct(NetworkFirewallPolicy):  Snapshot Test 1`] = `
Object {
  "Resources": Object {
    "TestPolicy14DF854C3": Object {
      "Properties": Object {
        "FirewallPolicy": Object {
          "StatefulDefaultActions": Array [
            "statefulDefaultActions",
          ],
          "StatefulEngineOptions": Object {
            "RuleOrder": "statefulEngineOptions",
          },
          "StatefulRuleGroupReferences": Array [
            Object {
              "Priority": 123,
              "ResourceArn": "resourceArn",
            },
          ],
          "StatelessCustomActions": Array [
            Object {
              "ActionDefinition": Object {
                "PublishMetricAction": Object {
                  "Dimensions": Array [
                    Object {
                      "Value": "CustomValue",
                    },
                  ],
                },
              },
              "ActionName": "actionName",
            },
          ],
          "StatelessDefaultActions": Array [
            "statelessDefaultActions",
          ],
          "StatelessFragmentDefaultActions": Array [
            "statelessFragmentDefaultActions",
          ],
          "StatelessRuleGroupReferences": Array [
            Object {
              "Priority": 123,
              "ResourceArn": "resourceArn",
            },
          ],
        },
        "FirewallPolicyName": "TestFirewallPolicy1",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestFirewallPolicy1",
          },
        ],
      },
      "Type": "AWS::NetworkFirewall::FirewallPolicy",
    },
    "TestPolicyCC05E598": Object {
      "Properties": Object {
        "FirewallPolicy": Object {
          "StatefulEngineOptions": Object {
            "RuleOrder": "STRICT_ORDER",
          },
          "StatefulRuleGroupReferences": Array [
            Object {
              "Priority": 123,
              "ResourceArn": "arn:aws:network-firewall:us-east-1:222222222222:stateful-rulegroup/TestGroup",
            },
          ],
          "StatelessDefaultActions": Array [
            "aws:forward_to_sfe",
          ],
          "StatelessFragmentDefaultActions": Array [
            "aws:forward_to_sfe",
          ],
        },
        "FirewallPolicyName": "TestFirewallPolicy",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestFirewallPolicy",
          },
        ],
      },
      "Type": "AWS::NetworkFirewall::FirewallPolicy",
    },
  },
}
`;
